### PR TITLE
Refactor Output Observers

### PR DESCRIPTION
**Pull request by domenukk**
This gets rid of the hacks in StdOutObserver et al.

@VTCAKAVSMoACE I got issues with type inference atm, but we should wait here for your AT PR #767 to land, before it makes no sense.

@andreafioraldi what do you think

### CODE REVIEW

File: fuzzers/backtrace_baby_fuzzers/command_executor/src/main.rs [hunk 1]
```diff
@@ -108,12 +108,11 @@ pub fn main() {
...
     }
 
-    let mut executor = MyExecutor { shmem_id }.into_executor(
-        tuple_list!(observer, bt_observer),
-        None,
-        None,
-        Some("ASANBacktraceObserver".to_string()),
-    );
+    // Hint: use .add_stderr or .add_stdout to add more observers here
+    let output_observers = OutputObservers::stderr(&bt_observer);
```
Comment (domenukk) [inline fuzzers/backtrace_baby_fuzzers/command_executor/src/main.rs:112]: New usage is like this

Comment (domenukk) [reply fuzzers/backtrace_baby_fuzzers/command_executor/src/main.rs]: nope, dropped that

**Comment by domenukk** [comment]
I feel like the AsanBacktraceObserver should stack on top of the Stdout observer, but at least this works.
Python doesn't support stdout and stderr handling yet

**Comment by andreafioraldi** [comment]
i don't like the stdout methods in the observer trait

**Comment by domenukk** [comment]
Better than manually entering stings

**Comment by andreafioraldi** [comment]
> Better than manually entering strings

yes but i think these methods can be moved out to another trait (e.g. CanObserveStdout) and in the executor put a bound like OT: ObserversTuple + CanObserveStdoutTuple

**Comment by domenukk** [comment]
What would be the Benefit of that?

**Comment by domenukk** [comment]
Think from a user point of view it's the right thing, for sure an update over the previous version, feel free to refactor again later

**Pull request merged**